ncclient/operations/edit.py

```diff
@@ -66,6 +66,12 @@ def request(self, config, format='xml', target='candidate', default_operation=No
         if format == 'text':
             config_text = sub_ele(node, "config-text")
             sub_ele(config_text, "configuration-text").text = config
+        if format == 'url':
+            if "://" in config:
```

Comment:
You should probably use https://docs.python.org/3/library/urllib.parse.html to check for URL validity.